network-api/networkapi/mozfest/migrations/0017_update_wide_images.py

```diff
@@ -0,0 +1,65 @@
+# # Generated by Django 3.0.14 on 2021-08-10 18:19
+
+from django.db import migrations
+import wagtail.core.blocks
+import wagtail.core.blocks.static_block
+import wagtail.core.fields
+import wagtail.images.blocks
+import wagtailmedia.blocks
+
+
+from networkapi.mozfest.models import MozfestPrimaryPage
+
+
+
+def update_annotated_image_blocks(qs):
+    # Loop through all pages
+    for page in qs:
+        print("\t", page)
+        # If this page doesn't have a Image that needs update, don't save it.
+        # This will speed up how fast this task can run.
+        needs_saving = False
+        # Only loop through streamfield data if there's a `body` field on the page
+        if page.body:
+            # Looping through raw_data is a List of blocks.
+            for block in page.body.raw_data:
+                # If there is a type in the blocks, and the block type is a image,
+                # check if the image has the wide setting on
+                if 'type' in block and block['type'] == "image":
+                    print(block)
+                    # If so, set the new image_width value to wide
+                    if 'wide_image' in block['value'] and block['value']['wide_image'] == True:
+                        print("Found a wide image, updating.")
+                        block['value']['image_width'] = "wide"
+                        needs_saving = True
+
+            if needs_saving:
+                # If page is published already, continue to publish it.
+                # Otherwise just save a revision for draft history.
+                if page.live:
+                    print("\t\tPage is live. Publish it")
+                    page.save_revision().publish()
+                else:
+                    print("\t\tPage is draft.")
+                    page.save_revision()
+
+def loop_through_pages_with_image_blocks(apps, schema):
+    # Only look through pages that ACTUALLY USE the Annotated Image Block streamfield.
+
+    pages_with_image_blocks = [
+        MozfestPrimaryPage.objects.all(),
+    ]
```

Comment:
this creates a single-element list, with a queryset as first and only element, but then you take that queryset back out in the for loop, so that's several lines of code that ends up just being `MozfestPrimaryPage.objects.all()` again =)

Reply:
Hi @Pomax! Thanks for the catch, I have tried to implement the suggested change from above, where instead of creating a list,  we use `    for page in MozfestPrimaryPage.objects.all():`, however, I seem to be running into a interesting error when doing so. 

When I run `inv test-python` using the suggested change, the test suite then returns this error: 

`django.db.utils.ProgrammingError: column wagtailcore_page.title_en does not exist`
`LINE 1: ...re_page"."locale_id", "wagtailcore_page"."title", "wagtailco...`
`HINT:  Perhaps you meant to reference the column "wagtailcore_page.title".`

This error is returned even if I comment everything out underneath the "for" loop, and adding a print statement instead. 

However, if I add this line above the for loop like so: 
`    if MozfestPrimaryPage.objects.all().count() != 0:`
`       for page in MozfestPrimaryPage.objects.all():`

The test suite then runs OK, and the migration properly updates wide images when run. 

I have tried to troubleshoot for a while now but have not been able to find the reason behind this, however I do see that while running the test suite, if allowed to run all the way through, seems to run some SQL statements that add the "title_en" column to the test DB:

<img width="757" alt="image" src="https://user-images.githubusercontent.com/18314510/129284796-218e6f48-c837-4fba-97f6-26220bb98548.png">

Then after using [this WIP PR](https://github.com/mozilla/foundation.mozilla.org/blob/f92b8175ce710aa06d6e578791a0a35483853c79/network-api/networkapi/wagtailpages/migrations/0020_migrate_quote_blocks.py#L49) for reference regarding custom migrations, I noted that we were filtering for objects with the "locale_id" of 1, for English like so:

`    pages_with_quote_blocks = [`
       ` BlogPage.objects.filter(locale_id=1),`
       ` PrimaryPage.objects.filter(locale_id=1),`
      `  ModularPage.objects.filter(locale_id=1),`
      `  MozfestPrimaryPage.objects.filter(locale_id=1)`
  `  ]` 

I found that in the error that is being produced from the testing suite, it appears that the "locale_id" column **does** exist in the DB, so I tried to use the filter statement as well, but the same error persists. 

I was wondering if this has been something that you have run into before when making a custom migration? 

Thanks!

Reply:
Thanks for the detailed explanation Pomax! I have made the update, and am going to mark this conversation as resolved